### PR TITLE
[front] - chore: make InputBar sticky

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR makes the `InputBar` sticky and addresses various UI layout and overflow issues across multiple pages. It also streamlines scrolling and agent click handling in the assistant conversation component.

## Risk

Breaking UI

## Deploy Plan

- Deploy to `front-edge`
- Deploy `front`

### CODE REVIEW

**Comment by JulesBelveze** [comment]
Main challenge is that one the layout is slightly broken and a adding `sticky top-x` to the `InputBar` doesn't fix. We need to have one div taking the full width & height of the viewport minus the navigation bar. And adding a layout in this div.

The current implementation achieves the intent however it "brakes" the scrolling in other pages -> the scroll bar is not on the edge of the window (cf screenshot) 

Screenshot 2024-07-17 at 22 21 33

File: front/components/assistant/Sharing.tsx [hunk 1]
```diff
@@ -492,7 +492,7 @@ function SlackIntegrationDrawer({
         hasChanged={false}
         variant="side-sm"
       >
-        <div className="pt-8">
+        <div className="w-full max-w-4xl pt-8">
```
Comment (flvndvd) [inline front/components/assistant/Sharing.tsx:495]: 🤔

File: front/components/misc/DropzoneContainer.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export function DropzoneContainer({
   return (
     <div
       {...getRootProps()}
-      className="flex h-full w-full flex-col items-center"
+      className="flex h-full w-full flex-col items-center overflow-y-auto"
```
Comment (flvndvd) [inline front/components/misc/DropzoneContainer.tsx:51]: Not sure to understand why we need this one?

File: front/components/sparkle/AppLayout.tsx [hunk 3]
```diff
@@ -188,8 +185,8 @@ export default function AppLayout({
 
             <div
               className={classNames(
-                "flex h-[calc(100%-5rem)] w-full flex-col",
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:191]: 😘

File: front/components/assistant/Sharing.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/assistant/Sharing.tsx:495]: Can we add pt-8 in here:
https://github.com/dust-tt/dust/blob/dff04e365bd2f35fa9a22f7221de6ad679e567a2/front/components/sparkle/AppLayout.tsx#L192

Comment (flvndvd) [reply front/components/assistant/Sharing.tsx:495]: ☝️ Meant for other places where `pt-8` makes sense.

**Pull request closed without merging**